### PR TITLE
dist/tools/uhcpd: add an option to bind to device

### PR DESCRIPTION
**Pull request by cladmi**
Adds '--bind-to-device' option to binds uhcpd listening socket to a device.
It allows running multiple uhcpd and so multiple border router on the same
computer.

This is used for IoT-LAB.

This option requires 'root' permissions as I currently didn't find an
alternative.

### CODE REVIEW

**Comment by cladmi** [comment]
This is related to issue https://github.com/RIOT-OS/RIOT/issues/5647 but only works as root.

File: dist/tools/uhcpd/uhcpd.c [hunk 1]
```diff
@@ -144,3 +161,13 @@ int udp_sendto(uint8_t *buf, size_t len, uint8_t *dst, uint16_t dst_port, uhcp_i
 
     return res;
 }
+
+/* Requires to be 'root', I didn't find another solution for the moment */
+static void bind_to_device(int sock, const char *interface)
+{
+    if (setsockopt(sock, SOL_SOCKET, SO_BINDTODEVICE,
+                interface, IF_NAMESIZE) < 0 ) {
```
Comment (aabadie) [inline dist/tools/uhcpd/uhcpd.c]: `interface` should be aligned on the `sock` parameter in the line above.

**Comment by aabadie** [comment]
ACK when Murdock is happy and the branch is rebased.

**Comment by aabadie** [comment]
[Murdock](https://ci.riot-labs.de/RIOT-OS/RIOT/6011/9fae3014e72213101a88ebaf2ea585b9827548d9/output.html#static-tests) also asks for squashing

**Comment by miri64** [comment]
Feature freeze -> postponed

**Comment by aabadie** [comment]
@miri64, I think this can be kept for the release, if @cladmi just squashes the commit in time.

**Comment by miri64** [comment]
If it get's merged before feature freeze (which will probably some time around tomorrow), who am I to judge.

**Comment by aabadie** [comment]
ping @cladmi

**Comment by cladmi** [comment]
Murdock wasn't happy but because of another failed test (dtls).
I rebased on master to re-run build.

**Comment by miri64** [comment]
the dtls test was fixed ;-)

**Comment by kYc0o** [comment]
Alles gut, @aabadie can you ACK?

**Comment by aabadie** [comment]
So go !

**Pull request merged**